core/transaction-pool/graph/src/pool.rs

```diff
@@ -115,109 +133,39 @@ pub struct Pool<B: ChainApi> {
 impl<B: ChainApi> Pool<B> {
 	/// Imports a bunch of unverified extrinsics to the pool
 	pub fn submit_at<T>(&self, at: &BlockId<B::Block>, xts: T, force: bool)
-		-> Result<Vec<Result<ExHash<B>, B::Error>>, B::Error>
+		-> Result<impl Future<Output=Vec<Result<ExHash<B>, B::Error>>>, B::Error>
```

Comment:
Having `Result<Future, Error>` seems a bit weird, if we ever extend the range of async APIs further it will complicate the returned errors. Maybe let's just return a `Future<Output=Result<..>>`?

Reply:
Just double-checking all your comments. This has been fixed by switching from `Result<Future<Vec<Result>>>` to `Future<Result<Vec<Result>>>`